docs/models/blockstates/introduction.md

```diff
@@ -19,28 +19,29 @@ As an example, let's take a look at the vanilla `oak_log.json`:
 
 Here we define 3 variant strings, and for each we use a certain model, either the upright log and the sideways log (rotated in the y direction or not). These variants will define the look of a log depending on the property `axis`.
 
-A blockstate always has to be defined for all possible variant strings. When you have many properties, this results in lots of possible variants, as every combination of properties must be defined. In Minecraft 1.8's blockstate format, you have to define every string explicitly, which leads to long, complicated files. Starting from Minecraft 1.9, Mojang also introduced the "multipart" format. You can find a definition of its format on the [wiki][].
+A blockstate must be defined for all possible variants that invoke a change in the current model. Any property not specified in the JSON will not have any bearing to determine the current model (e.g. `waterlogged` has no effect on how a model might look). As each property adds a factor of two to the amount of states, there can be a number of possible variants needing a defined model.
```

Comment:
I dislike that first sentence still, it contradicts itself just after. If you *must* define all variants, how can you not specify them as well? It's also mixing terms a bit, the individual things inside the blockstate are called *variants*, so calling them blockstates here is strange.

Additionally, each property does not add a factor of two, it depends how many valid values that property has. That last sentence is probably just unnecessary.

Reply:
I'll reword that then quickly. My change in that sentence was that _must_ points to that sentence end where for all possible variants that invoke a change in the current model. It would be more applicable to say property, but I'll change the sentence such that it clarifies the point more.